lib/dry/system/component.rb

```diff
@@ -47,49 +48,78 @@ class Component
       #   @return [Object#call] component's loader object
       attr_reader :loader
 
+      # Returns a component with a namespace and path provided from a file found within
+      # the given component dirs. If no file is found, a component is returned without
+      # these attributes.
+      #
+      # @return [Dry::System::Component]
       # @api private
-      def self.new(*args, &block)
-        cache.fetch_or_store([*args, block].hash) do
```

Comment:
@solnic I couldn't figure out why Component needed a cache, since every component should only get initialized once over the lifecycle of any given container, either as the auto-registrar runs, or as each component is lazily resolved. In the name of simplifying things while I worked, I removed the cache, but I'd love to know if there's any reason you think it needs to remain in place.

Reply:
@timriley well, I don't remember 🤷🏻 it's possible that in the past it was needed and then it wasn't...and I just missed that fact.